File: scripts/spline/common/postprocess.R
library(STAR)

args <- commandArgs(trailingOnly = TRUE)
inpath = paste(args[1], "out", "data", sep="/")
outpath = paste(args[1], "results", "postprocessed.RData", sep="/")

read_rdata <- function(file) {
	r_data <- load(file)
	x <- get(r_data)
	return(x)
}

predict.one.var <- function(result, var) {
	pred = quickPredict(result$gss, var)

	x = pred$xx
	mean = pred$est.mean
	sd = pred$est.sd

	if (is.function(result$inv.rnfun[[var]])) {
		x = result$inv.rnfun[[var]](x)
	}

	return(list(x=x, mean=mean, sd=sd))
}

predict.one.result <- function(file) {
	result = read_rdata(file)
	time = "time"
	if (result$reference == "multi") {
		time = "timetoevt"
	}

	vars = c(time, "r.timeSinceLastSpike")
	if (result$group != "all") {
		vars <-c(vars, "r.nearest")
	}


	res = list(index=result$index, reference=result$reference, group=result$group)

	for (var in vars) {
		res[[var]] = predict.one.var(result, var)
		}
	return(res)
	}

infiles <- list.files(path=inpath, pattern=".R", full.names=T, all.files=T)

predictions = lapply(infiles, predict.one.result)

outpath = "~/ens/data/analyses/spline/batch-5/results/postprocessed.RData"
save(predictions, file=outpath)

File: scripts/spline/common/postprocess.jl
using DrWatson
@quickactivate :ens

using RData
using DataFrames
using DataFramesMeta
using Spikes
using Arrow

include(srcdir("spline", "model_summaries.jl"))


inpath = "$(ARGS[1])results/postprocessed.RData"
outpath = "$(ARGS[1])results/result.arrow"


data = RData.load(inpath)["predictions"]

result = DataFrame(index1=Float64[],
				   index2=Float64[], 
				   group=String[],
				   reference=String[],
				   variable=String[],
				   x=Vector{Float64}[],
				   mean=Vector{Float64}[],
				   sd=Vector{Float64}[])

for row in data
	variables  = filter(x -> (x != "index" && x!= "group" && x != "reference"), 
						row.index2name)

	index1 = row["index"][1]
	index2 = length(row["index"]) > 1 ? row["index"][2] : NaN
	group = row["group"]
	reference = row["reference"]

	for v in variables
		push!(result, (index1, index2, group, reference, v, row[v]["x"], row[v]["mean"], row[v]["sd"]))
	end
end

Arrow.write(outpath, result)
